src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+// import edu.wpi.first.wpilibj2.*;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+/** Add your docs here. */
+// Put methods for controlling this subsystem
+// here. Call these from Commands.
+public class BatterySubsystem extends SubsystemBase {
+
+  private ShuffleboardTab btTab;
+  private NetworkTableEntry sbVoltage;
+  private NetworkTableEntry sbInCurrent;
+  private NetworkTableEntry sbTimer;
+  private NetworkTableEntry sbTimerChange;
+  private Timer timer;
+
+  private void init() {
+    timer.reset();
+    btTab = Shuffleboard.getTab("Battery");
+    sbVoltage = btTab.add("Battery Voltage", 0).withSize(2, 2).withPosition(0, 0).getEntry();
+    sbInCurrent = btTab.add("Input Current", 0).withSize(2, 2).withPosition(2, 0).getEntry();
+    sbTimer = btTab.add("Timer", 0).withSize(2, 2).withPosition(0, 2).getEntry();
+    sbTimer = btTab.add("Change Timer", 0).withSize(2, 2).withPosition(2, 2).getEntry();
+    timer.start();
+  }
+
+  public BatterySubsystem() {
+    init();
+  }
+
+  public void periodic() {
+    sbVoltage.setDouble(getVoltage());
+    sbInCurrent.setDouble(getInputCurrent());
+    sbTimer.setDouble(getTimer());
+    sbTimerChange.setBoolean(
```

Comment:
We want to warn if any of the three conditions happen:
1.  High current usage for some amount of time
2. Any usage for a longer time
3. Dangerously low voltage. 

It would also be great to have a "yellow zone" of optional alert, and then have a more forceful, "danger zone" alert.
Maybe even disable the robot in the danger zone?

Maybe yellow zone is 1 or 2?  and danger zone is passing 3?  Not sure what is best.

Reply:
I should note, not sure that one can disable the robot programatically.  But I feel like we need to try and consider a way to make it impossible to ignore, if we can

Is there something more annoying that we can do in the danger zone?  Continuous beeping on the robot?

Reply:
Completed, yellow and red zone exist only in the actual code, the simulation only uses 1 marker for checking times